File: Array Sorting Algorithms/bubble-sort/comb-sort.py
def comb_sort(arr):
    n = len(arr)
    gap = n
    shrink_factor = 1.3
    swapped = True

    while gap > 1 or swapped:
        gap = max(1, int(gap / shrink_factor))
        swapped = False

        for i in range(n - gap):
            if arr[i] > arr[i + gap]:
                arr[i], arr[i + gap] = arr[i + gap], arr[i]
                swapped = True

# Example usage:
arr = [64, 34, 25, 12, 22, 11, 90]
comb_sort(arr)
print("Sorted array:", arr)

File: Array Sorting Algorithms/bubble-sort/comb-sort.java
import java.util.Arrays;

public class CombSort {
    public static void combSort(int[] arr) {
        int n = arr.length;
        int gap = n;
        double shrinkFactor = 1.3;
        boolean swapped = true;

        while (gap > 1 || swapped) {
            gap = Math.max(1, (int) (gap / shrinkFactor));
            swapped = false;

            for (int i = 0; i + gap < n; i++) {
                if (arr[i] > arr[i + gap]) {
                    int temp = arr[i];
                    arr[i] = arr[i + gap];
                    arr[i + gap] = temp;
                    swapped = true;
                }
            }
        }
    }

    // Example usage:
    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90};
        combSort(arr);
        System.out.println("Sorted array: " + Arrays.toString(arr));
    }
}
